src/main/learnocaml_client.ml

```diff
@@ -812,11 +812,11 @@ module Grade = struct
          (Printf.eprintf "Results NOT saved to server (deadline expired)\n";
           Lwt.return 1)
        else
-         match token with
-         | Some token ->
+         match eo.submit, token with
```

Comment:
Just out of curiosity, could you elaborate a bit (in a PR comment) on the addition of this condition "eo.submit = true" ?

Reply:
See related commit message (commit messages are better than PR comments IMHO, since they live in the project rather than on a proprietary platform)
This is actually the CLI flag that is the negation of `--dry-run`